triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerLogContext.scala

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger
+
+import com.daml.lf.engine.trigger.Runner.Implicits._
+import com.daml.logging.LoggingContextOf
+import com.daml.logging.LoggingContextOf.label
+import com.daml.logging.entries.LoggingValue.OfString
+import com.daml.logging.entries.{LoggingEntries, LoggingValue}
+
+import java.util.UUID
+
+object ToLoggingContext {
+  implicit def `TriggerLogContext to LoggingContextOf[Trigger]`(implicit
+      triggerContext: TriggerLogContext
+  ): LoggingContextOf[Trigger] =
+    LoggingContextOf
+      .withEnrichedLoggingContext(
+        label[Trigger],
+        "trigger" -> LoggingValue.Nested(LoggingEntries(triggerContext.entries: _*)),
+      )(triggerContext.loggingContext)
+      .run(identity)
+}
+
+final case class TriggerLogContext private (
+    loggingContext: LoggingContextOf[Trigger],
```

Comment:
In typical usage, this logging context will be the base one on which the rest of this class then maintains the "trigger" log object